tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -0,0 +1,767 @@
+	.reconfig = bap_unicast_server_cb_reconfig,
+	.qos = bap_unicast_server_cb_qos,
+	.enable = bap_unicast_server_cb_enable,
+	.start = bap_unicast_server_cb_start,
+	.metadata = bap_unicast_server_cb_metadata,
+	.disable = bap_unicast_server_cb_disable,
+	.stop = bap_unicast_server_cb_stop,
+	.release = bap_unicast_server_cb_release,
+};
+
+static void *ascs_ase_control_test_suite_setup(void)
+{
+	struct ascs_ase_control_test_suite_fixture *fixture;
+
+	fixture = malloc(sizeof(struct ascs_ase_control_test_suite_fixture));
+	zassume_not_null(fixture);
+
+	fixture->ase_cp = ascs_get_attr(BT_UUID_ASCS_ASE_CP, 1);
+	fixture->ase_snk_1 = ascs_get_attr(BT_UUID_ASCS_ASE_SNK, 1);
+
+	bt_bap_unicast_server_register_cb(&unicast_server_cb);
+
+	return fixture;
+}
+
+static void ascs_ase_control_test_suite_before(void *f)
+{
+	struct ascs_ase_control_test_suite_fixture *fixture = f;
+
+	ARG_UNUSED(fixture);
+
+	bt_conn_ref_fake.custom_fake = conn_ref_custom_fake;
+	bt_conn_get_info_fake.custom_fake = conn_get_info_custom_fake;
+	bt_gatt_foreach_attr_type_fake.custom_fake = gatt_foreach_attr_type_custom_fake;
+	bt_pacs_cap_foreach_fake.custom_fake = pacs_cap_foreach_custom_fake;
+
+	bap_unicast_server_cb_config_fake.return_val = 0;
+	bap_unicast_server_cb_reconfig_fake.return_val = 0;
+	bap_unicast_server_cb_qos_fake.return_val = 0;
+	bap_unicast_server_cb_enable_fake.return_val = 0;
+	bap_unicast_server_cb_start_fake.return_val = 0;
+	bap_unicast_server_cb_metadata_fake.return_val = 0;
+	bap_unicast_server_cb_disable_fake.return_val = 0;
+	bap_unicast_server_cb_stop_fake.return_val = 0;
+	bap_unicast_server_cb_release_fake.return_val = 0;
+}
+
+static void ascs_ase_control_test_suite_after(void *f)
+{
+	ascs_cleanup();
+}
+
+static void ascs_ase_control_test_suite_teardown(void *f)
+{
+	struct ascs_ase_control_test_suite_fixture *fixture = f;
+
+	bt_bap_unicast_server_unregister_cb(&unicast_server_cb);
+
+	free(fixture);
+}
+
+ZTEST_SUITE(ascs_ase_control_test_suite, NULL, ascs_ase_control_test_suite_setup,
+	    ascs_ase_control_test_suite_before, ascs_ase_control_test_suite_after,
+	    ascs_ase_control_test_suite_teardown);
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_read_state_idle)
+{
+	uint8_t data_expected[] = {
+		0x01,   /* ASE_ID */
+		0x00,   /* ASE_State = Idle */
+	};
+	ssize_t ret;
+
+	ret = fixture->ase_snk_1->read(&fixture->conn, fixture->ase_snk_1, NULL, 0, 0);
+	zassert_false(ret < 0, "attr->read returned unexpected (err 0x%02x)",
+		      BT_GATT_ERR(ret));
+
+	expect_single_call_bt_gatt_attr_read(&fixture->conn, fixture->ase_snk_1, NULL, 0, 0,
+					     data_expected, ARRAY_SIZE(data_expected));
+}
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_control_operation_zero_length_write)
+{
+	uint8_t buf[] = {};
+	ssize_t ret;
+
+	ret = fixture->ase_cp->write(&fixture->conn, fixture->ase_cp, (void *)buf, 0, 0, 0);
+	zassert_true(ret < 0, "ase_cp_attr->write returned unexpected (err 0x%02x)",
+		     BT_GATT_ERR(ret));
+}
```

Comment:
What about a test with `NULL` buf? Or is that expected never to happen?

Reply:
I would say that this is expected never to happen as per GATT API.